pages/qr.tsx

```diff
@@ -5,10 +5,30 @@ import { Text, Card, Box } from '~ui/core'
 import { isFormal } from '~lib/config'
 import { ArrowsLeft, ArrowsRight } from '~ui/anicons'
 import { MobileApp } from '~ui/layouts/MobileApp'
+import { useRouter } from 'next/router'
 
 export default function QRCodePage() {
   const videoEl = React.useRef<HTMLVideoElement>()
 
+  const router = useRouter()
+
+  const paramsRef = React.useRef<object>({})
+
+  paramsRef.current = {
+    name: router.query.name?.toString(),
+    phone: router.query.phone?.toString(),
+    address: router.query.address?.toString(),
+  }
```

Comment:
It's strange that we're not experiencing the issues in https://github.com/railslove/rcvr-app/blob/master/pages/checkin.tsx#L43-L45

Reply:
In checkin.tsx we are accessing that information from within the render method. So it is always up to date. 

In qr.tsx we are accessing it from within a callback function that was created in the effect hook. I think the callback is not up to date. I hit similar issues and it can be related to https://stackoverflow.com/questions/57847594/react-hooks-accessing-up-to-date-state-from-within-a-callback.